lib/puppet/provider/package/gem.rb

```diff
@@ -124,13 +130,20 @@ def install(useversion = true)
           command << source
         else
           # interpret it as a gem repository
-          command << "--source" << "#{source}" << resource[:name]
+          command << "--source" << "#{source}"
```

Comment:
Copying my comment from the other PR. This change looks unrelated.

Reply:
If the purpose of specifying —no-rdoc  etc is to avoid generating/installing gem documentation (and avoid the time necessary to do so), then it is a logical error to conditionally not do so based upon the souce parameter. 

I’ll remove this change tomorrow,  double check my work and submit the correction as part of another PR.

Reply:
Ah, Charlie helped me understand the issue (with just `<< resource[:name]`) ... will adjust.

Reply:
Reverted the unrelated change involving `:source` in the first commit.